lib/WUI/link_content/basic_gets.cpp

```diff
@@ -249,4 +253,17 @@ JsonResult get_job(size_t resume_point, JsonOutput &output) {
     // clang-format on
 }
 
+JsonResult get_storage(size_t resume_point, JsonOutput &output) {
+    // Keep the indentation of the JSON in here!
+    // clang-format off
+    JSON_START;
+        JSON_OBJ_START
+            JSON_FIELD_STR("path", "/usb") JSON_COMMA;
+            JSON_FIELD_STR("type", "USB") JSON_COMMA;
+            JSON_FIELD_BOOL("available", opendir("/usb"));
```

Comment:
The `opendir` is missing its closedir (+style-wise, this should probably be `opendir(..) != nullptr`, to avoid the implicit pointer->bool conversion).

I'd suggest pulling this out into a helper function (otherwise the macros around the JSON will get unhappy about a local-scoped variable) in an anonymous namespace.

As another point. Would something more lightweight work? I think `stat` for example won't allocate dynamically, unlike `opendir`, but I'm not sure if it does what we need.